src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/resources/stylesheet.css

```diff
@@ -425,77 +402,89 @@ div.table-tabs > button.table-tab {
    background: #4D7A97;
    color: #FFFFFF;
 }
-
-.row-color th,
-.alt-color th {
-    font-weight:normal;
-}
-.overview-summary td, .member-summary td, .type-summary td,
-.use-summary td, .constants-summary td, .deprecated-summary td,
-.requires-summary td, .packages-summary td, .provides-summary td,
-.uses-summary td, .system-properties-summary td {
+.two-column-summary {
+    display: grid;
+    grid-template-columns: minmax(15%, max-content) minmax(15%, auto);
+}
+.three-column-summary {
+    display: grid;
+    grid-template-columns: minmax(10%, max-content) minmax(15%, max-content) minmax(15%, auto);
+}
+.four-column-summary {
+    display: grid;
+    grid-template-columns: minmax(10%, max-content) minmax(10%, max-content) minmax(10%, max-content) minmax(10%, auto);
+}
+@media screen and (max-width: 600px) {
+    .two-column-summary {
+        display: grid;
+        grid-template-columns: 1fr;
+    }
+}
+@media screen and (max-width: 800px) {
+    .three-column-summary {
+        display: grid;
+        grid-template-columns: minmax(10%, max-content) minmax(25%, auto);
+    }
+    .three-column-summary .col-last {
+        grid-column-end: span 2;
+    }
+}
+@media screen and (max-width: 1000px) {
+    .four-column-summary {
+        display: grid;
+        grid-template-columns: minmax(15%, max-content) minmax(15%, auto);
+    }
+}
+.summary-table > div {
     text-align:left;
-    padding:0px 0px 12px 10px;
+    padding: 8px 3px 3px 7px;
```

Comment:
General comment: should we be using some font-related unit such as `em` `ex` instead of absolute units like `px` ?

Reply:
I haven't delved into the pros and cons of different CSS measurement units, but I can do that. For this change, the purpose was to leave spacings unchanged, so experimenting with different units was not something I wanted to explore.

Reply:
Definitely not in this round, but maybe sometime. I would look at the units through the lens of absolute sizes, like `pt` and `px` compared to font-relative sizes, like `em` and `ex` etc.